pkg/reconciler/knativeserving/common/certs.go

```diff
@@ -54,6 +55,8 @@ func CustomCertsTransform(instance *servingv1alpha1.KnativeServing, log *zap.Sug
 			if err := scheme.Scheme.Convert(deployment, u, nil); err != nil {
 				return err
 			}
+			// Avoid superfluous updates from converted zero defaults
+			u.SetCreationTimestamp(metav1.Time{})
```

Comment:
I'm curious if these would now also be fixed due to the three-way-merge semantics? I.e., do we need to manually set them still?

Reply:
We're relying on [this logic](https://github.com/knative/serving-operator/blob/master/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go#L353-L356) to actually remove the field. Without it, we attempt to apply the `Time`'s zero-value set by the typed object.

Reply:
I view it as a tradeoff for being able to treat any type in the manifest as `Unstructured`. I would prefer not to special-case any fields at this time, if we can avoid it.

And though I share your concern about the non-obviousness, keep these in mind:
* It's only a problem if you convert the unstructured to a structured type and back again. Most transforms (currently) don't do this.
* The only risk to not clearing the zeroed fields is executing unnecessary updates; annoying and slow, but logically harmless.
* It will be very obvious to anyone looking at the log that unnecessary updates are occurring because we now log the diff showing why we're updating any resource. I would expect this to trigger the transform author, so it may be less non-obvious than we think.

Reply:
Sounds good. Thanks.